libs/librepcb/common/attributes/attributetype.cpp

```diff
@@ -69,6 +69,15 @@ const AttributeUnit* AttributeType::getUnitFromString(const QString& unit) const
         QString(tr("Unknown unit of attribute type \"%1\": \"%2\"")).arg(mTypeName, unit));
 }
 
+bool AttributeType::isUnitAvailable(const AttributeUnit* unit) const noexcept
+{
+    if (mAvailableUnits.isEmpty()) {
+        return (unit == nullptr);
+    } else {
+        return mAvailableUnits.contains(unit);
+    }
```

Comment:
Actually the class `AttributeType` requires some refactoring anyway, so I think i will keep the current implementation and postpone the proposed improvements. I'm sorry ;)